flexmeasures/ui/templates/base.html

```diff
@@ -519,15 +520,18 @@
                             nearEnd,
                             end,
                         );
-                    }
-
+                    } 
+                    
                     // No use looking for data in years outside timerange of sensor data
                     picker.setOptions({
                         dropdowns: {
                             minYear: start.getFullYear(),
                             maxYear: end.getFullYear(),
                         },
                     });
+
+                    // Persist the range on the datepicker UI so correct range is shown when the user opens the datepicker
+                    picker.setDateRange(decodeURIComponent('{{event_starts_after}}'), decodeURIComponent('{{event_ends_before}}'));
```

Comment:
Refreshing the page inadvertently leads to extending the date range by 1 day. I suggest to subtract 1 day from the end time, as follows:

```suggestion
                    var end_time = '{{event_ends_before}}';
                    end_date = new Date(end_time);
                    end_date.setDate(end_date.getDate() - 1);
                    picker.setDateRange(decodeURIComponent('{{event_starts_after}}'), decodeURIComponent(toIsoString(end_date)));
```

Reply:
I don't know if this makes sense, but maybe we should not use `decodeURIComponent('{{event_starts_after}}')`, but `start`, which is the one coming from the response. And line 513 might take care of the extra day issue.

Reply:
I am currently trying out @Flix6x suggestion, will revert, and also try your suggestion also @nhoening

Reply:
@Flix6x's suggestion worked, but the date selection doesn’t persist when I leave and return to the page. However, the issue with adding an extra day seems to be resolved, as confirmed by the graph.

@nhoening, for some reason, the start and end variables aren't holding the correct values. They keep returning the 6th instead of the session date. I’ve been reviewing the code for a while but haven’t figured out the cause yet.

Reply:
@Flix6x, when I leave the page, the end_date variable becomes undefined. I’m not exactly sure what's causing this at the moment.

Reply:
Also thinking about your previous comment, the response date range should be the same with what is on the session right?

Reply:
You an look at the endpoint. I would have expected it is the date range of the returned data, however the endpoint decided what start and end should be.

Reply:
thats what im asking though, should the endpoint be responsible for deciding, or should the session handle it?

Reply:
> @Flix6x, when I leave the page, the end_date variable becomes undefined. I’m not exactly sure what's causing this at the moment.

my work around for this is the below

`  var end_time = '{{event_ends_before}}';
                    var new_end_date = new Date(end_time);
                    // new_end_date.setDate(new_end_date.getDate() - 1);
                    picker.setDateRange(decodeURIComponent('{{event_starts_after}}'), decodeURIComponent(toIsoString(new_end_date)));`
                    
This works and it persists but now I realize that some code somewhere is reducing the start date 😆, I'll look into the code again and see if I can find something.

Reply:
The start date reduction has to do with the timezone being switched to UTC somewhere. So when the start datetime is midnight on e.g. day 7 in Amsterdam, it becomes 10 PM on day 6 in UTC (same time, different timezone, and a different day). That is, 2024-09-07T00:00+02 becomes 2024-09-06T22:00Z (the Z stands for +00).

We should try to avoid the conversion to UTC, but I'm not sure where it happens. Alternatively, we could make sure to convert back to the user's timezone before passing the datetime's date to the datepicker.

Reply:
So for this, the approach I took was to deduct it from the server side before sending it over